q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,322 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import yaml
+from glob import glob
+import pandas as pd
+from os.path import basename, isdir, splitext
+
+from q2_metadata.normalization._norm_messages import warningsCollection, errorsCollection
+
+
+class Rules(object):
+    """Collect the rules of one current variable rules.
+
+    Parameters
+    ----------
+    variable_rules_fp : str
+        Path to one variable's rules yaml file.
+
+    Attributes
+    ----------
+    variable : str
+        Name of the variable which rules are parsed and
+        that is the yaml file name without the .yml extension.
```

Comment:
if the variable name is expected to be the same as the filename, would it make sense to not require this be passed in?

Reply:
maybe I'm wrong but to this `Rules()` class, the variable name contained in `variable` is an *attribute* that indeed the file name (without the path directories and without yaml), and not a *parameter*. Only the file path is passed in.
Or do you mean that we could keep the entire file path as variable name? That could be good and the actual variable name-only string could be extracted later when needed. For now it is not effectively used (i.e proposed to the user) but in the final machine-readable output, it will be valuable to have the path anyway.

Reply:
UPDATE: the variable name is collected during parsing of the .yml files in the `parse_variables_rules()` method of the `RulesCollection()` class (like this: `variable = splitext(basename(variable_rules_fp))[0]`). 

It is important to have because used:
- as dict key in the main variables_rules object (see `variables_rules = rules.parse_variables_rules()` in the main script `q2_metadata/_normalize()`).
- for the error messages to be concise (assuming the users understand the the yml files are named per variables' names...) - for this, it is passed as parameters to `Rules()`instantiations and no longer is an attribute.